curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderSelector.java

```diff
@@ -341,11 +342,41 @@ public Participant getLeader() throws Exception
 
     static Participant getLeader(CuratorFramework client, Collection<String> participantNodes) throws Exception
     {
+        Participant result = null;
+        
         if ( participantNodes.size() > 0 )
         {
-            return participantForPath(client, participantNodes.iterator().next(), true);
+            Iterator<String> iter = participantNodes.iterator();
+            while ( iter.hasNext() )
+            {
+                
+                try
+                {
+                    result = participantForPath(client, iter.next(), true);
```

Comment:
Yep, fair call, it's probably more Java like to just return. I've probably been writing too much C code lately!